Adding erc20 deploy contract page (#173)
casper-network/docs: source/docs/casper/dapp-dev-guide/tutorials/erc20/deploy.md

# Deploy the Contract
import useBaseUrl from '@docusaurus/useBaseUrl';

> Now that you have implemented the smart contract for ERC20 and it's time to deploy. Deploying the ERC20 contract is similar to deploying other smart contracts, while only the WASM files and parameters will differ. Refer [Deploying Contracts](/docs/dapp-dev-guide/deploying-contracts#advanced-deployments) section to learn more about overall contract deployment.

Reviewer: Use ERC-20 instead of ERC20 in all places.

> Now that you have implemented the smart contract for ERC20 and it's time to deploy. Deploying the ERC20 contract is similar to deploying other smart contracts, while only the WASM files and parameters will differ. Refer [Deploying Contracts](/docs/dapp-dev-guide/deploying-contracts#advanced-deployments) section to learn more about overall contract deployment.

Reviewer: Suggestion to apply these edits:
"Now that you have implemented the smart contract for ERC-20, it's time to deploy it to the blockchain. Deploying the ERC-20 contract is similar to deploying other smart contracts, while only the WASM files and parameters will differ. Refer to the [Deploying Contracts](/docs/dapp-dev-guide/deploying-contracts#advanced-deployments) section to learn more about overall contract deployment."

Let's dive into the deployment process.

### Pre-requisites

- Set up your machine as per the [prerequisites](/docs/workflow/setup)

> - You need to have a pre-created `Casper Signer Account` with the required amount of tokens to perform the deploy

Reviewer: Point to the Signer guide: https://casper.network/docs/workflow/signer-guide

> - Account should contain enough `CASPR tokens` to proceed with the deploy. These will use to pay for the transactions on the Casper Network (involving ERC-20 tokens).

Reviewer 2: CSPR

> - Account should contain enough `CASPR tokens` to proceed with the deploy. These will use to pay for the transactions on the Casper Network (involving ERC-20 tokens).

Reviewer: Remove "." at the end

> - Generated `Public Key` and `Private Key` to perform the deploy and related actions

Reviewer: What about moving this with the Signer account point?

> - Installed [casper client](/dapp-dev-guide/tutorials/counter/setup) to interact with the network

Reviewer: Let's capitalize Casper

## Basic Flow
You will go through the following basic steps to execute the ERC20 contract deployment on the Casper network.

> 

Reviewer: Capitalize Network. Also, you can say: "Here are the basic steps to deploy the ERC-20 contract on the Casper Network."

> <img src={useBaseUrl("/image/tutorials/erc-20/erc20-deply-flow.png")} alt="erc20-deploy-flow" width="500"/>

Reviewer: Add ERC-20 in the image. Also, you could say, "Query Contract" instead.

## Clone the ERC20 Contract
This step includes cloning and preparing the ERC20 contract for the deployment.

> 1. Clone the contract from the repo

Reviewer: I'd say "repository" or GitHub and add a link to it.

```bash
git clone https://github.com/casper-ecosystem/erc20.git
```
2. Move to the newly created folder and compile the contract to create the target WASM file 
```bash
cd erc20
make prepare
```

> 3. Builds and verifies the compiled contract

Reviewer: Say "Build and verify" to match the other points.

```bash
make test
```

## Get an IP Address from a Testnet Peer 
Use the [acquire node address](/docs/workflow/setup#acquire-node-address-from-network-peers) section to get a node-ip-address. We use [peers](https://testnet.cspr.live/tools/peers) on testnet since we are deploying to the testnet. Select a peer address from the list and do the address format as below,
:::note
Substitute '7777' in place of '35000' to get the correct command for the deploy. If the selected peer is unresponsive, pick a different peer and try again

> :::

Reviewer: Please simplify this as such:
"[Acquire a node address](/docs/workflow/setup#acquire-node-address-from-network-peers) from the Testnet and use port '7777' instead of '35000' to send your deploy. If the selected peer is unresponsive, pick a different peer and try again."

Testnet peers page sample:

> <img src={useBaseUrl("/image/tutorials/erc-20/testnet-peers.png")} alt="erc20-deploy-flow" width="800"/>

Reviewer: Nice to have: it would be good to mask out the key in the top right, with a program such as Snagit or Skitch. Let me know if you have questions about this.

## View the Network Status

> This query capture any information related to the state of the blockchain at the specific time denoted by state root hash. To view this, you need to have the state root hash and the account hash. Use the below commands to obtain those:

Reviewer: I would simplify:
"This query captures information related to the state of the blockchain at the specific time denoted by the state root hash. You need to have the state root hash and the account hash to run the query."

> **Get State Root Hash**

Reviewer: "Getting the State Root Hash" (following the styling guide)

> This marks a snapshot of the network state at a moment of time.

Reviewer: Use "moment in time"

```bash
casper-client get-state-root-hash --node-address http://<HOST:PORT>
```

> **Get Account Hash**

Reviewer: Getting the Account Hash (following the styling guide)

> This is the identification of your account.

Reviewer: Re-write: "To get the account hash, run the following command and supply the path to your Public Key in Hex format."

```bash

> casper-client account-address --public-key "/home/ubuntu/public_key_hex"

Reviewer: Use a generic path instead.

```

> **Query the Network State**

Reviewer: Use gerund according to the style guide: "Querying..."

> Use the below command template to query the network status with regards to your account.

Reviewer: "Use the command template below"

```bash
casper-client query-state \
--node-address http://<HOST:PORT> \
--state-root-hash [STATE_ROOT_HASH] \
--key [ACCOUNT_HASH]
```

> - `<HOST:PORT>`: Use the [node-ip-address](../erc20/deploy#get-an-ip-address-from-a-testnet-peer) taken from a testnet peer.

Reviewer: Move this above, in the instructions on how to use the command. 
Capitalize Testnet. 
Say Node IP address rather than "node-ip-address".

## Deploy the Contract
Now you can deploy the contract on the network and check how it behaves. Use the following command template to deploy the contract:
```bash
casper-client put-deploy \
    --node-address http://<HOST:PORT> \
    --chain-name [NETWORK_NAME]] \
    --secret-key [PATH_TO_YOUR_KEY]/secret_key.pem \
    --payment-amount [AMOUNT] \
    --session-path [WASM_FILE_PATH]/[File_Name].wasm
```
- `NETWORK_NAME`: Use the relevant network name. Here we use 'testnet'
- `PATH_TO_YOUR_KEY`: Replace this with the actual path of your secret key 

> - `PAYMENT_AMOUNT`: Gas amount in tokens send for the execution. If there are no adequate tokens, deploy will not execute and will return an error

Reviewer: You need to mention the faucet somewhere and point to this page: https://casper.network/docs/workflow/token-transfer#2-the-faucet.
Also re-write: "Gas amount in tokens needed for contract execution. If there are no adequate tokens, the deploy will not execute and return an error"

> - `WASM FILE PATH`: The session-path argument should point to wherever you compiled erc20 WASM file on your machine 

Reviewer: Re-write: "The session-path argument should point to the location of your compiled ERC-20 WASM file."

:::note

> If you are performing the deploy in the mainnet, it's recommended to try several put deploys on the testnet or local machine to figure out the exact amount required for that deploy. Refer [note about gas price](/docs/dapp-dev-guide/deploying-contracts#a-note-about-gas-prices) section to understand more about payment amounts and gas price adjustments

Reviewer: Re-write: "If you are performing the deploy on the Mainnet, we recommend trying several deploys on the Testnet to understand the exact amount required for the deploy. Refer to the [note about gas price](/docs/dapp-dev-guide/deploying-contracts#a-note-about-gas-prices) to understand more about payment amounts and gas price adjustments." Add a colon. Also, do not mention a local network, unless you want to specify that the local network needs to have the same configuration as the Mainnet (otherwise the gas price would be different).

:::

Find the sample command below:

```bash
casper-client put-deploy \
--node-address http://95.216.24.237:7777 \
--chain-name casper-test \
--secret-key "/home/ubuntu/secret_key.pem" \
--payment-amount 1000000 \
--session-path "/home/ubuntu/erc20/target/wasm32-unknown-unknown/release/erc20_test.wasm"
```

> ## Query the Network Status

Reviewer: Use gerunds in the titles, according to the style guide.

> Now you have deployed the contract to the network and the status of the network has changed. To view the network status you need to get the newest state root hash because it has changed with the deploy. Account hash remains the same since you are using the same account. Follow [View network state](../erc20/deploy#view-the-network-status) section to execute this step with the new state root hash.

Reviewer: Simplify wording, remove the first sentence, and use: "You need to get the newest state root hash to view the network status because it has changed with the deploy. The account hash remains the same since you are using the same account. Follow [View network state](../erc20/deploy#view-the-network-status) to execute this step with the new state root hash."

## Verify the Deploy

> Now we can verify the applied deploy using the `get deploy` command. This will output the details of the applied deploy.

Reviewer: Use "you" instead of "we".

```bash
casper-client get-deploy \
--node-address http://<HOST:PORT> [DEPLOY_HASH]
```

## Querying with Arguments

> We checked the query state at a global level in [previous query the network](../erc20/deploy#view-the-network-status) step. In this step, we will narrow down the context and check the status specifically for a certain area. You will use the details inside [erc20 contract](https://github.com/casper-ecosystem/erc20/blob/master/example/erc20-token/src/main.rs) section to derive arguments.

Reviewer: Recommending to re-write this and removing the first sentence: 
"This step will narrow down the context and check the status of a specific entry point. You will use the details inside the [ERC-20 contract](https://github.com/casper-ecosystem/erc20/blob/master/example/erc20-token/src/main.rs) to derive arguments."

> We checked the query state at a global level in [previous query the network](../erc20/deploy#view-the-network-status) step. In this step, we will narrow down the context and check the status specifically for a certain area. You will use the details inside [erc20 contract](https://github.com/casper-ecosystem/erc20/blob/master/example/erc20-token/src/main.rs) section to derive arguments.

Reviewer: What do you mean by "certain area"? Could you be more specific? Do you refer to entry points?

Author: Yes, it should be entry point - Thanks Iulia

:::

> Use below command template to query with arguments:

Reviewer: "Use the command template below to query the network state:"

```bash
casper-client query-state \
--node-address http://<HOST:PORT> \
--state-root-hash [STATE_ROOT_HASH] \
--key [ACCOUNT_HASH] \
-q "[CONTRACT_NAME/ARGUMENT]"
```

## Sample Deploy on Testnet

> The following steps will guide you through the series of above-described steps with actual values and results.

Reviewer: I am not sure about this, Buddhika. People will start copying and pasting.

Reviewer: If you really want to keep this, then create your own values such as 9999999999....9999999.

Reviewer: Re-write "guide you through the **process** with actual values and results".

Author: > If you really want to keep this, then create your own values such as 9999999999....9999999.

I am not much bias on this point @ipopescu,  after your thoughts, I also felt like better to remove this part, shall we ask from @GloreenS , @darthsiroftardis, and @deuszex as well?

**Clone the ERC20 contract**

```bash
git clone https://github.com/casper-ecosystem/erc20.git
```
**Get an IP Address from a Testnet Peer** 

Use [peers](https://testnet.cspr.live/tools/peers) site to get the node ip address.
Eg:  http://95.216.24.237:7777

**View the network status**

Here is the command:
```bash
casper-client query-state \

> --key account-hash-7f4bf39A311a7538d8C91BB86C71DF774023e16bc4a70ab7e4e8AE77DbF2Ef53 \

Reviewer: I would not include the account address.

--node-address http://95.216.24.237:7777 \

> --state-root-hash E5B679BD1562fE6257257F5f969A79482E8DCEBBD501501BfA6d5844b61cBE3f

Reviewer: Use generic values... I am not sure about adding such details in the docs.

```

Query Result:

> This contains the network state before the deploy. You can see the `named-key` field is empty since we haven't done any deploy on the network yet.

Reviewer: "This **result** contains .."

```bash

> {

Reviewer: Collapse the result when it is very long. See https://casper.network/docs/workflow/querying#obtaining-the-global-state-root-hash.

  "id": 401803927542812599,
  "jsonrpc": "2.0",
  "result": {
    "api_version": "1.4.3",
    "merkle_proof": "[25564 hex chars]",
    "stored_value": {
      "Account": {
        "account_hash": "account-hash-7f4bf39A311a7538d8C91BB86C71DF774023e16bc4a70ab7e4e8AE77DbF2Ef53",
        "action_thresholds": {
          "deployment": 1,
          "key_management": 1
        },
        "associated_keys": [
          {
            "account_hash": "account-hash-7f4bf39A311a7538d8C91BB86C71DF774023e16bc4a70ab7e4e8AE77DbF2Ef53",
            "weight": 1
          }
        ],
        "main_purse": "uref-C051e7EC16e08Def8b556F9CD0E18FE701C89dC0cED3DAd7b65107285da198DD-007",
        "named_keys": []
      }
    }
  }
}
```

**Deploy the contract**

> The deploy command:

Reviewer: "Deploy the contract with this command:"

Reviewer: I know this is an example, but I think all commands need to have generic paths, otherwise, people will copy and paste them without thinking about the details.

```bash
casper-client put-deploy \
--node-address http://95.216.24.237:7777 \
--chain-name casper-test \
--secret-key "/home/ubuntu/secret_key.pem" \
--payment-amount 1000000 \
--session-path "/home/ubuntu/erc20/target/wasm32-unknown-unknown/release/erc20_test.wasm"
```

Result:

This command execution will output the `deploy_hash` of the applied deploy. We can use the deploy_hash to get the details of the deploy.
```bash
{
  "id": 931694842944790108,
  "jsonrpc": "2.0",
  "result": {
    "api_version": "1.4.3",
    "deploy_hash": "b00E59f8aBA5c7aB9bfA496ae4Aec7Ec8A9F0227179F5F09AcA594335F62dc1f"
  }
}
```

> **View the deploy details**

Reviewer: Use gerund here

> You can view the details of the applied deploy using below command:

Reviewer: "the command below"

```bash
casper-client get-deploy \

> --node-address http://95.216.24.237:7777 \

Reviewer: Use generic node addresses, otherwise people will just copy and paste.

b00E59f8aBA5c7aB9bfA496ae4Aec7Ec8A9F0227179F5F09AcA594335F62dc1f
```

Result:

> This contains the header, payment and session details along with the 

Reviewer: Sentence not completed. What did you try to say here?

```bash

> {

Reviewer: Add a section to collapse the result. The How To guides use this.

                  "transform": "Identity"
                },
                {
                  "key": "balance-98d945f5324F865243B7c02C0417AB6eaC361c5c56602FD42ced834a1Ba201B6",
                  "transform": "Identity"
                },
                {
                  "key": "balance-62f7Fe1ceCB1a4C600fFA791479ce52Fb8cBDA408815F4Dd1B1E0D82e704579a",
                  "transform": "Identity"
                },
                {
                  "key": "balance-98d945f5324F865243B7c02C0417AB6eaC361c5c56602FD42ced834a1Ba201B6",
                  "transform": {
                    "WriteCLValue": {
                      "bytes": "00",
                      "cl_type": "U512",
                      "parsed": "0"
                    }
                  }
                },
                {
                  "key": "balance-62f7Fe1ceCB1a4C600fFA791479ce52Fb8cBDA408815F4Dd1B1E0D82e704579a",
                  "transform": {
                    "AddUInt512": "100000"
                  }
                }
              ]
            },
            "error_message": "Out of gas error",
            "transfers": []
          }
        }
      }
    ]
  }
}

```

> **Query with Arguments**

Reviewer: Use gerund: Querying...

# Deploy the Contract
import useBaseUrl from '@docusaurus/useBaseUrl';

Now that you have implemented the smart contract for ERC-20, it's time to deploy it to the blockchain. Deploying the ERC-20 contract is similar to deploying other smart contracts, while only the WASM files and parameters will differ. Refer [Deploying Contracts](/docs/dapp-dev-guide/deploying-contracts#advanced-deployments) section to learn more about overall contract deployment.

Let's dive into the deployment process.

### Pre-requisites

- Set up your machine as per the [prerequisites](/docs/workflow/setup)

> - You need to have a pre-created [Casper signer account](https://casper.network/docs/workflow/signer-guide) with the required amount of tokens to perform the deploy

Reviewer 3: This sentence implies that the account must be created with the Casper signer, when thats not really true. As long as they have an Account it would be fine. 

```suggestion
- You need to have a pre-created [Account](https://casper.network/docs/workflow/setup#setting-up-an-account) with the required amount of tokens to perform the deploy
```

>   - You will receive the `Public Key` and `Private Key` along with the Casper signer account which are mandatory items for the future deploys

Reviewer 3: ```suggestion
  - You will receive the `Public Key` and `Private Key` corresponding to the `Account` which are mandatory items for the future deploys
```

**Getting State Root Hash**

This marks a snapshot of the network state at a moment in time.
```bash
casper-client get-state-root-hash --node-address http://<HOST:PORT>
```
- `<HOST:PORT>`: Use the [Node IP address](../erc20/deploy#get-an-ip-address-from-a-testnet-peer) taken from a Testnet peer.

**Getting Account Hash**

To get the account hash, run the following command and supply the path to your Public Key in Hex format.
```bash
casper-client account-address --public-key "[PATH_TO_YOUR_KEY]/public_key_hex"
```
**Querying the Network State**

Use the command template below to query the network status with regards to your account.
```bash
casper-client query-state \
--node-address http://<HOST:PORT> \
--state-root-hash [STATE_ROOT_HASH] \
--key [ACCOUNT_HASH]
```

## Deploy the Contract
Now you can deploy the contract on the network and check how it behaves. Use the following command template to deploy the contract:
```bash
casper-client put-deploy \
    --node-address http://<HOST:PORT> \
    --chain-name [NETWORK_NAME]] \
    --secret-key [PATH_TO_YOUR_KEY]/secret_key.pem \
    --payment-amount [AMOUNT] \
    --session-path [WASM_FILE_PATH]/[File_Name].wasm
```

> - `NETWORK_NAME`: Use the relevant network name. Here we use 'testnet'

Reviewer 3: ```suggestion
- `NETWORK_NAME`: Use the relevant network name. Here we use 'casper-test'
```

Testnet's network name is `casper-test`

- `PATH_TO_YOUR_KEY`: Replace this with the actual path of your secret key 
- `PAYMENT_AMOUNT`: Gas amount in tokens needed for contract execution. If there are no adequate tokens, the deploy will not execute and return an error
- `WASM FILE PATH`: The session-path argument should point to the location of your compiled ERC-20 WASM file

:::note

> If you are performing the deploy in the Mainnet, we recommend trying several put deploys on the Testnet to understand the exact amount required for that deploy. Refer to the [note about gas price](/docs/dapp-dev-guide/deploying-contracts#a-note-about-gas-prices) to understand more about payment amounts and gas price adjustments

Reviewer 3: Please add a note, that we currently do not refund any tokens as part of a deploy. If the deployment costed say 1 CSPR but you sent the deploy with 10 CSPR, the system will not return the extra 9 CSPR

:::

Find the sample command below:

```bash
casper-client put-deploy \
--node-address http://95.216.24.237:7777 \
--chain-name casper-test \
--secret-key "/home/ubuntu/secret_key.pem" \
--payment-amount 1000000 \
--session-path "/home/ubuntu/erc20/target/wasm32-unknown-unknown/release/erc20_test.wasm"
```

## Querying the Network Status
You need to get the newest state root hash to view the network status because it has changed with the deploy. The account hash remains the same since you are using the same account. Follow [View network state](../erc20/deploy#view-the-network-status) to execute this step with the new state root hash.

## Verifying the Deploy

> Now you can verify the applied deploy using the `get deploy` command. This will output the details of the applied deploy.

Reviewer 3: ```suggestion
Now you can verify the applied deploy using the `get-deploy` command. This will output the details of the applied deploy.
```

```bash
{
  "id": 8893083480582974265,
  "jsonrpc": "2.0",
  "result": {
    "api_version": "1.4.3",
    "deploy": {
      "approvals": [
        {
          "signature": "[130 hex chars]",
          "signer": "017B8CE645c7285689F79281C5AB60348d14D3a9d1A419f3441A993402dd81dbE2"
        }
      ],
      "hash": "b00E59f8aBA5c7aB9bfA496ae4Aec7Ec8A9F0227179F5F09AcA594335F62dc1f",
      "header": {
        "account": "017B8CE645c7285689F79281C5AB60348d14D3a9d1A419f3441A993402dd81dbE2",
        "body_hash": "7B5c246dbC76D5F9D8Af4CB49F1c69AFa6bE9dAB2f8d624153c2e682dB469DbD",
        "chain_name": "casper-test",
        "dependencies": [],
        "gas_price": 1,
        "timestamp": "2022-01-04T10:47:55.437Z",
        "ttl": "30m"
      },
      "payment": {
        "ModuleBytes": {
          "args": [
            [
              "amount",
              {
                "bytes": "03a08601",
                "cl_type": "U512",
                "parsed": "100000"
              }
            ]
          ],
          "module_bytes": ""
        }
      },
      "session": {
        "ModuleBytes": {
          "args": [],
          "module_bytes": "[417800 hex chars]"
        }
      }
    },

>     "execution_results": [

Reviewer 3: Please mention that if this field is represented as `"execution_results":[]` means that the deploy hasn't executed yet. Depending on the network, there will be some time before this field contains the results of the execution.

Author: @darthsiroftardis , please explain the second part of this sentence, is that mean different networks may take different times to load the execution result?

                },
                {
                  "key": "hash-9824d60dC3A5c44A20b9FD260a412437933835B52Fc683d8AE36e4ec2114843e",
                  "transform": "Identity"
                },
                {
                  "key": "balance-98d945f5324F865243B7c02C0417AB6eaC361c5c56602FD42ced834a1Ba201B6",
                  "transform": "Identity"
                },
                {
                  "key": "balance-62f7Fe1ceCB1a4C600fFA791479ce52Fb8cBDA408815F4Dd1B1E0D82e704579a",
                  "transform": "Identity"
                },
                {
                  "key": "balance-98d945f5324F865243B7c02C0417AB6eaC361c5c56602FD42ced834a1Ba201B6",
                  "transform": {
                    "WriteCLValue": {
                      "bytes": "00",
                      "cl_type": "U512",
                      "parsed": "0"
                    }
                  }
                },
                {
                  "key": "balance-62f7Fe1ceCB1a4C600fFA791479ce52Fb8cBDA408815F4Dd1B1E0D82e704579a",
                  "transform": {
                    "AddUInt512": "100000"
                  }
                }
              ]
            },

>             "error_message": "Out of gas error",

Reviewer 3: Could we please provide an example of a successful deployment

# Contract Deployment
import useBaseUrl from '@docusaurus/useBaseUrl';

Now that you have implemented the smart contract for ERC-20, it's time to deploy it to the blockchain. Deploying the ERC-20 contract is similar to deploying other smart contracts, while only the WASM files and parameters will differ. Refer to the [Deploying Contracts](/docs/dapp-dev-guide/deploying-contracts#advanced-deployments) section to learn more about overall contract deployment.

Let's dive into the deployment process.

### Pre-requisites

- Set up your machine as per the [prerequisites](/docs/workflow/setup)

> - Ensure you have a pre-created [Account](https://casper.network/docs/workflow/setup#setting-up-an-account) with the required amount of tokens to perform the deploy

Reviewer 3: ```suggestion
- Ensure you have an [Account](https://casper.network/docs/workflow/setup#setting-up-an-account), the associated key pair,  and the required amount of tokens to perform the deploy
```

>   - You will receive the *Public Key* and *Private Key* along with the Casper signer account. Those will use to query the network

Reviewer 3: This line can be removed

- Ensure your [account](https://testnet.cspr.live/tools/faucet) contains enough `CSPR Tokens` to perform the execution. 

>   -Tokens will use to pay for the transactions on the Casper Network (involving ERC-20 tokens). Follow the [transfer tokens](https://casper.network/docs/workflow/token-transfer#2-the-faucet) guide to learn more about token transferring on the Casper Testnet

Reviewer 3: This line is very confusing, what are we trying to point out here?

Author: It tells why tokens are used in this context - I have altered the wording with new changes

- Install [Casper client](/dapp-dev-guide/tutorials/counter/setup) to interact with the network

## Basic Flow
Here are the basic steps to deploy the ERC-20 contract on the Casper Network.

<img src={useBaseUrl("/image/tutorials/erc-20/erc20-deploy-flow.png")} alt="erc20-deploy-flow" width="600"/>

## Cloning the ERC-20 Contract
This step includes cloning and preparing the ERC-20 contract for the deployment.
1. Clone the ERC-20 contract from the repository
```bash
git clone https://github.com/casper-ecosystem/erc20.git
```
2. Move to the newly created folder and compile the contract to create the target WASM file 
```bash
cd erc20

> make prepare

Reviewer 3: This is incorrect, this will just setup the dependencies for the contract and won't actually build the contract. 

```suggestion
make prepare
make build-contracts
```

The `make build-contracts` command will actually build the WASM
